reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -0,0 +1,524 @@
+"""Iterative refinement with Bayesian expectation maximization."""
+
+import numpy as np
+from simSPI.transfer import eval_ctf
+
+
+class IterativeRefinement:
+    """Iterative refinement with max likelihood estimation.
+
+    Parameters
+    ----------
+    map_3d_init : arr
+        Initial particle map.
+        Shape (n_pix, n_pix, n_pix)
+    particles : arr
+        Particles to be reconstructed.
+        Shape (n_particles, n_pix, n_pix)
+    ctf_info : list of dicts
+        Each dict contains CTF k,v pairs per particle.
+            Shape (n_particles,)
+
+    References
+    ----------
+    1. Nelson, P. C. (2019). Chapter 12 : Single Particle
```

Comment:
Use this updated ref from the url. Sorry I had an out of date one.

Nelson, P. C. (2021). Physical Models of Living Systems new chapter: Single Particle Reconstruction in Cryo-electron Microscopy. Retrieved from https://repository.upenn.edu/physics_papers/656